docs/src/lib/types.md

```diff
@@ -57,6 +57,52 @@ The `RepeatedVector` and `StackedVector` types are subtypes of `AbstractVector`
 with the exception that they are read only. Note that they are not exported and should not be constructed directly,
 but they are columns of a `DataFrame` returned by `stackdf` and `meltdf`.
 
+## [The design of handling of columns of a `DataFrame`](@id man-columnhandling)
+
+When a `DataFrame` is constructed columns are copied by default. You can disable
+this behavior by setting `copycols` keyword argument to `false`. The exception is
+if an `AbstractRange` is passed as a column, then it is always collected to a `Vector`.
+
+Also functions that transform a `DataFrame` to produce a new `DataFrame` perform a copy of the columns,
+unless they are passed `copycols=false` (available only for functions
+that could perform a transformation without copying the columns). Examples of such functions are [`vcat`](@ref),
+[`hcat`](@ref), [`filter`](@ref), [`dropmissing`](@ref), [`join`](@ref), `getindex`,
+[`copy`](@ref) or the [`DataFrame`](@ref) constructor mentioned above.
+
+On the contrary, functions that create a view of a `DataFrame` *do not* by definition make copies of
+the columns, and therefore require particular caution. This includes `view`, which returns
+a `SubDataFrame` or a `DataFrameRow`, and `groupby`, which returns a `GroupedDataFrame`.
+
+A partial exception to this rule are the [`stackdf`](@ref) and [`meltdf`](@ref) functions which
+create a `DataFrame` that contains views of the columns from the source `DataFrame`.
+
+In-place functions whose names end with `!` (like `sort!` or [`dropmissing!`](@ref),
+`setindex!`, `push!`, `append!`) may mutate the column vectors of the `DataFrame` they take
+as an argument. These functions are safe to call due to the rules described above,
+*except* when a view of the `DataFrame` is in use (via a `SubDataFrame`, a `DataFrameRow`
+or a `GroupedDataFrame`). In the latter case, the view might become corrupted,
+which make trigger errors, silently return invalid data or even cause Julia to crash.
+
+It is possible to have a direct access to a column `col` of a `DataFrame` `df`
+(e.g. this can be useful in performance critical code to avoid copying),
+using one of the following methods:
+
+* via the `getproperty` function using the syntax `df.col`;
+* via the `getindex` function using the syntax `df[:col]`;
+* by creating a `DataFrameColumns` object using the [`eachcol`](@ref) function;
+* by storing the reference to the column before creating a `DataFrame` with `copycols=false`;
```

Comment:
Does `@view df[:col, :]` also get a reference to the column?

Reply:
indirectly yes, ie. `@view df[:, :col]` is a view of the column, so you could extract a column via `parent` function. I added this case (but it is probably not very likely to happen in practice).

Reply:
Why not overload this case to return a column?
that would be `Base.view(df::DataFrame, ::Colon, col::Symbol) = getproperty(df, col)`,
right?

`@view df[:, :col]` would likely endup being my preferred style if it were permitted.
Since it feels maximally like dealing with an `Array`

Reply:
The problem is that then `@view` would not be type stable. I.e. if you wrote `@view df[rows, :col]` you would get either a view or a vector.

Note that Base also returns a view when `:` is used:
```
julia> x = [1,2,3]
3-element Array{Int64,1}:
 1
 2
 3

julia> @view x[:]
3-element view(::Array{Int64,1}, :) with eltype Int64:
 1
 2
 3
```

You can think of `view(df, :, :col)` as an equivalent of writing `view(df[:col], :)`.

Reply:
Are you sure what I proposed wouldn't be type stable?
 -   `@view df[rows, :col]` i.e. `view(df, rows, :col)`  would return a `SubDataFrame`  if `rows` was a `AbstractArray`,
 -  `@view df[:, :col]` i.e. `view(df, :, :col)`  would return a `Vector`, and this is fine, because `Colon` is its own type.

Sounds type stable to me. Output type seems like it only depends on the input type.
Though I'm reconsidering its wisdom, since it would be confusing to have that return a different kind of object(`Vector` vs `AbstractDataFrame` subtype) to what `@view df[1:end, :col]` returns.
---

What I actually care about is that 
 `@view df[:, :col]` (i.e. `view(df, :, :col)`), is noncopying.

Which we have right now, right?

Reply:
> Which we have right now, right?

Right, `@view df[:, :col]` is non copying and will be non-copying (it returns a view).

You are right about type stability :), I was thinking exactly about the difference between  `:` and `1:end` though (which you have spotted anyway 😄).

Also note that `@view df[rows, :col]` returns a `SubArray` not `SubDataFrame` (which is I guess what you would have expected). `SubDataFrame` would be returned if we had multiple columns selected (not one).

Reply:
Ah right,
well a SubArray and a Full array are very similar.
As such my orginal suggestion actually might make sense then.

Though there _should_ be negligable performance difference between a full array view, and the array itself, and there should be no functional difference.

So 🤷‍♂️

Reply:
I agree - that it why I would leave it as is for now (as this is not directly changed in this PR anyway).